glotaran/builtin/megacomplexes/decay/util.py

```diff
@@ -326,10 +327,15 @@ def retrieve_irf(dataset_model: DatasetModel, dataset: xr.Dataset, global_dimens
             model_axis=dataset.coords[model_dimension].values,
         ).data,
     )
+
     center = irf.center if isinstance(irf.center, list) else [irf.center]
     width = irf.width if isinstance(irf.width, list) else [irf.width]
     dataset["irf_center"] = ("irf_nr", center) if len(center) > 1 else center[0]
     dataset["irf_width"] = ("irf_nr", width) if len(width) > 1 else width[0]
+
+    if irf.shift is not None:
+        dataset["irf_shift"] = (global_dimension, [center[0] - p.value for p in irf.shift])
```

Comment:
It's hard (especially without unit test) to predict the types here, could you indicate in a comment the structure and types you expect so that when typing is added or unit tests are created this can be taken into account?

Reply:
Just looked it up `irf.center` is always a `list[Parameter]` and `shift` is `list[Parameter] | None`

```python

@model_item(
    properties={
        "center": List[Parameter],
        ...
        "shift": {"type": List[Parameter], "allow_none": True},
        ...
    },
    has_type=True,
)
class IrfMultiGaussian:
```